src/test/java/bio/terra/app/controller/DatasetsApiControllerTest.java

```diff
@@ -1,153 +1,192 @@
 import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.IngestRequestValidator;
 import bio.terra.service.dataset.exception.DatasetDataException;
+import bio.terra.service.filedata.FileService;
+import bio.terra.service.job.JobService;
 import java.util.List;
 import java.util.UUID;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.web.servlet.MockMvc;
 
-@RunWith(SpringRunner.class)
-@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
 @ActiveProfiles({"google", "unittest"})
-@AutoConfigureMockMvc
-@Category(Unit.class)
+@ContextConfiguration(classes = {DatasetsApiController.class, GlobalExceptionHandler.class})
+@Tag("bio.terra.common.category.Unit")
+@WebMvcTest
 public class DatasetsApiControllerTest {
-  @MockBean private DatasetService datasetService;
   @Autowired private MockMvc mvc;
+  @MockBean private JobService jobService;
+  @MockBean private DatasetRequestValidator datasetRequestValidator;
+  @MockBean private DatasetService datasetService;
+  @MockBean private IamService iamService;
+  @MockBean private FileService fileService;
+  @MockBean private AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @MockBean private AssetModelValidator assetModelValidator;
+  @MockBean private IngestRequestValidator ingestRequestValidator;
+  @MockBean private DataDeletionRequestValidator dataDeletionRequestValidator;
+  @MockBean private DatasetSchemaUpdateValidator datasetSchemaUpdateValidator;
 
   private static final AuthenticatedUserRequest TEST_USER =
-      AuthenticatedUserRequest.builder()
-          .setSubjectId("DatasetUnit")
-          .setEmail("dataset@unit.com")
-          .setToken("token")
-          .build();
+      AuthenticationFixtures.randomUserRequest();
+  private static final String RETRIEVE_DATASET_ENDPOINT = "/api/repository/v1/datasets/{id}";
+  private static final DatasetRequestAccessIncludeModel INCLUDE =
+      DatasetRequestAccessIncludeModel.NONE;
   private static final String GET_PREVIEW_ENDPOINT =
-      "/api/repository/v1/datasets/{id}/data/{table}";
+      "%s/data/{table}".formatted(RETRIEVE_DATASET_ENDPOINT);
```

Comment:
Is there a reason to prefer `String.format()` over string concatenation?
```suggestion
      RETRIEVE_DATASET_ENDPOINT + "/data/{table}";
```

Reply:
Personal preference.  But I see it's more readable with concatenation, I took your suggestion.